typespec-extension/src/code-model-builder.ts

```diff
@@ -1689,275 +1707,250 @@ export class CodeModelBuilder {
+  private processBooleanSchemaFromSdkType(type: SdkBuiltInType, name: string): BooleanSchema {
     return this.codeModel.schemas.add(
-      new BooleanSchema(name, this.getDoc(type), {
-        summary: this.getSummary(type),
+      new BooleanSchema(name, this.getDoc(type.__raw as Scalar), {
+        summary: this.getSummary(type.__raw as Scalar),
       }),
     );
   }
 
-  private processArraySchema(type: ArrayModelType, name: string): ArraySchema {
-    const elementSchema = this.processSchema(type.indexer.value, name);
+  private processArraySchemaFromSdkType(type: SdkArrayType, name: string): ArraySchema {
+    const elementSchema = this.processSchemaFromSdkType(type.valueType, name);
     return this.codeModel.schemas.add(
-      new ArraySchema(name, this.getDoc(type), elementSchema, {
-        summary: this.getSummary(type),
+      new ArraySchema(name, this.getDoc(type.__raw as Type), elementSchema, {
+        summary: this.getSummary(type.__raw as Type),
       }),
     );
   }
 
-  private processDictionarySchema(type: RecordModelType, name: string): DictionarySchema {
-    const dictSchema = new DictionarySchema<any>(name, this.getDoc(type), null, {
-      summary: this.getSummary(type),
+  private processDictionarySchemaFromSdkType(type: SdkDictionaryType, name: string): DictionarySchema {
+    const dictSchema = new DictionarySchema<any>(name, type.details ? type.details : "", null, {
+      summary: type.description ? type.description : "",
     });
 
     // cache this now before we accidentally recurse on this type.
     if (!this.schemaCache.has(type)) {
       this.schemaCache.set(type, dictSchema);
     }
 
-    const elementSchema = this.processSchema(type.indexer.value, name);
+    const elementSchema = this.processSchemaFromSdkType(type.valueType, type.valueType.kind);
     dictSchema.elementType = elementSchema;
 
-    if (type.indexer.value.kind === "Union") {
-      dictSchema.nullableItems =
-        Array.from(type.indexer.value.variants.values()).findIndex((it) => isNullType(it.type)) >= 0;
-    }
+    dictSchema.nullableItems = type.nullableValues;
 
     return this.codeModel.schemas.add(dictSchema);
   }
 
-  private processChoiceSchema(
-    type: Enum,
+  private processChoiceSchemaFromSdkType(
+    type: SdkEnumType,
     name: string,
-    sealed: boolean,
   ): ChoiceSchema | SealedChoiceSchema | ConstantSchema {
-    const namespace = getNamespace(type);
-    const valueType =
-      typeof type.members.values().next().value.value === "number"
-        ? isAllValueInteger(Array.from(type.members.values()).map((it) => it.value as number))
-          ? this.integerSchema
-          : this.doubleSchema
+    const namespace = getNamespace(type.__raw as Enum);
+    const valueType = isSdkIntKind(type.valueType.kind)
+      ? this.integerSchema
+      : isSdkFloatKind(type.valueType.kind)
+        ? this.doubleSchema
         : this.stringSchema;
```

Comment:
Can we just do `processSchema` here, since TCGC already gives the type of the element?

Reply:
https://github.com/Azure/autorest.java/blob/main/typespec-extension/src/code-model-builder.ts#L1980-L1987
I was refering to our existing code on this logic. 
Good suggestion. Let me try to use `processSchema()` and test.